File: ruby/bob/bob.rb
class Bob
  def hey(remark)
    remark = remark.strip.split("\n").join(' ')
    if remark.match(/^\s*$/)
      'Fine. Be that way!'
    elsif remark.match(/^[A-Z,\s]+[!?]?$|^[A-Z\d,\s!@#$%^&*()]+!$/)
      'Whoa, chill out!'
    elsif remark[-1] == '?'
      'Sure.'
    else
      'Whatever.'
    end
  end
end

File: elixir/bob/bob.exs
defmodule Teenager do
  def hey(input) do
    cond do
      "" == String.strip(input) ->
        "Fine. Be that way!"
      input == String.upcase(input) and input != String.downcase(input) ->
        "Whoa, chill out!"
      String.ends_with?(input, "?") ->
        "Sure."
      true ->
        "Whatever."
    end
  end
end
